torch/utils/__init__.py

```diff
@@ -46,6 +46,27 @@ def swap_attr(name):
         setattr(t1, name, (getattr(t2, name)))
         setattr(t2, name, tmp)
 
+    def error_pre_hook(grad_outputs):
+        raise RuntimeError("Trying to execute AccumulateGrad node that was poisoned by swap_tensors")
+
+    def check_use_count(t, name='t1'):
+        use_count = t._use_count()
+        if use_count > 1:
+            if use_count == 2 and t.is_leaf:
+                accum_grad_node = torch.autograd.graph.get_gradient_edge(t).node
+                # Make sure that the accumulate_grad node was not lazy_init-ed by get_gradient_edge
+                if t._use_count() == 2:
+                    accum_grad_node.register_prehook(error_pre_hook)
+                else:
+                    raise RuntimeError(f"Expected use_count of {name} to be 1 or 2 with "
```

Comment:
Same for here - maybe something like "make sure you are not holding references to the tensor in other places"